### PR TITLE
Implement Countly

### PR DESCRIPTION
**Pull request by lorenzoPrimi**
Fixes https://github.com/ooni/probe/issues/1209

### CODE REVIEW

File: Podfile.lock [hunk 1]
```diff
@@ -1,9 +1,9 @@
 PODS:
-  - Crashlytics (3.14.0):
-    - Fabric (~> 1.10.2)
+  - Countly (20.04.2):
+    - Countly/Core (= 20.04.2)
+  - Countly/Core (20.04.2)
```
Comment (bassosimone) [inline Podfile.lock:4]: This line seems duplicated

File: ooniprobe/Engine/Engine.h [hunk 2]
```diff
@@ -15,6 +15,8 @@
 
 + (NSString*) versionMK;
 
++ (NSString*) getUUID;
```
Comment (bassosimone) [inline ooniprobe/Engine/Engine.h:18]: same comment as for Android: this is not a cached UUID, it's new each time. So you need to use another less ambiguous name here, like `newUUID4` and also cache it yourself.

File: ooniprobe/Utility/CountlyUtility.m [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+#import "SettingsUtility.h"
+#import "Engine.h"
+
+@implementation CountlyUtility
+
++ (void)initCountly {
+    CountlyConfig* config = CountlyConfig.new;
+    config.appKey = @"146836f41172f9e3287cab6f2cc347de3f5ddf3b";
+    config.host = NOTIFICATION_SERVER;
+    //Dabug builds
```
Comment (bassosimone) [inline ooniprobe/Utility/CountlyUtility.m:11]: please fix the spelling

File: ooniprobe/Utility/CountlyUtility.m [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+    #ifndef DEBUG
```
Comment (bassosimone) [inline ooniprobe/Utility/CountlyUtility.m:12]: here you want `#ifdef DEBUG`

File: Podfile.lock (hunk 1 above)
Comment (lorenzoPrimi) [reply Podfile.lock:4]: it's an auto generated file

File: ooniprobe/Utility/CountlyUtility.m [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+    //Only for testflights build
+    //config.pushTestMode = CLYPushTestModeTestFlightOrAdHoc;
+    config.features = @[CLYPushNotifications, CLYCrashReporting, CLYAutoViewTracking];
+    config.deviceID = [Engine getUUID];
+    config.requiresConsent = YES;
+    [Countly.sharedInstance startWithConfig:config];
+    [self reloadConsents];
+}
+
++ (void)reloadConsents{
```
Comment (bassosimone) [inline ooniprobe/Utility/CountlyUtility.m:25]: We should probably add consent for location here?

File: ooniprobe/Utility/SettingsUtility.h [hunk 6]
```diff
@@ -21,8 +21,9 @@
 
 + (BOOL)getSettingWithName:(NSString*)settingName;
 
-+ (NSString*)get_push_token;
++ (BOOL)isSendCrash;
```
Comment (bassosimone) [inline ooniprobe/Utility/SettingsUtility.h:24]: `isSendCrashEnabled`?

File: ooniprobe/View/Settings/SettingsTableViewController.m [hunk 7]
```diff
@@ -189,12 +185,20 @@ -(IBAction)setSwitch:(UISwitch *)mySwitch{
...
-        [self handleNotificationChanges];
-        [mySwitch setOn:FALSE];
+    if ([current isEqualToString:@"notifications_enabled"]){
+        if (mySwitch.on){
+            [Countly.sharedInstance giveConsentForFeature:CLYConsentPushNotifications];
+            [self handleNotificationChanges];
+            [mySwitch setOn:FALSE];
+        }
+        else
+            [Countly.sharedInstance giveConsentForFeature:CLYConsentPushNotifications];
```
Comment (bassosimone) [inline ooniprobe/View/Settings/SettingsTableViewController.m:195]: I don't understand this line ^

File: ooniprobe/View/TabsViewController.m [hunk 8]
```diff
@@ -24,20 +24,45 @@ -(void)viewDidAppear:(BOOL)animated{
...
+                                        }];
+        UIAlertAction* disableButton = [UIAlertAction
+                                    actionWithTitle:NSLocalizedString(@"Modal.ShareAnalytics.Disable", nil)
+                                    style:UIAlertActionStyleDefault
+                                    handler:^(UIAlertAction * action) {
+                                        [self setModalValue:NO
+                                                        key:@"send_analytics"
+                                                  popupName:ANALYTICS_POPUP];
+                                    }];
+        [MessageUtility alertWithTitle:NSLocalizedString(@"Modal.ShareAnalytics.Title", nil) message:NSLocalizedString(@"Modal.ShareAnalytics.Paragraph", nil) okButton:enableButton cancelButton:disableButton inView:self];
```
Comment (bassosimone) [inline ooniprobe/View/TabsViewController.m:58]: can you wrap this line? ^

**Pull request merged**